File: circularQueueLinkedList/circularQueue.cpp
#include "node.cpp"
class CircularQueue
{
	Node *front, *rear;
   public:
   	CircularQueue()
      {
      	front=rear=NULL;
      }
      CircularQueue(Node *front,Node *rear)
      {
      	this->front=front;
         this->rear=rear;
      }
      void enque()
      {
      	Node *nptr=new Node();
         if(nptr==NULL)
         {
         	cout<<endl<<"Overflow"<<endl;
            return;
         }
         if(isEmpty())
         	front=rear=nptr;
         else
         {
         	rear->setLink(nptr);
            rear=nptr;
         }
         rear->setLink(front);
         cout<<endl<<"Enter item: ";
         int item;
         cin>>item;
         rear->setData(item);
      }
      int delque()
      {
      	Node *temp=front;
         if(front==rear)
         	front=rear=NULL;
         else
         {
         	front=front->getLink();
            rear->setLink(front);
         }
         int item=temp->getData();
         delete temp;
         temp=NULL;
         return item;
      }
      void show()
      {
      	cout<<endl;
      	for(Node *ptr=front;ptr!=rear;ptr=ptr->getLink())
         	cout<<ptr->getData()<<" ";
         cout<<rear->getData()<<endl;
      }
      bool isEmpty()
      {
      	return front==NULL;
      }
};

File: circularQueueLinkedList/CircularQueue.java
import java.util.Scanner;
class CircularQueue
{
    private Node front,rear;
    public CircularQueue()
    {
        front=null;
        rear=null;
    }
    public CircularQueue(Node front,Node rear)
    {
        this.front=front;
        this.rear=rear;
    }
    public void enqueue()
    {
        Scanner sc=new Scanner(System.in);
        Node nptr=new Node();
        if(nptr==null)
        {
            System.out.println("\nOverflow");
            return;
        }
        System.out.print("\nEnter item: ");
        int item=sc.nextInt();
        nptr.setData(item);
        if(isEmpty())
        {
            front=nptr;
            rear=nptr;
        }
        else
        {
            rear.setLink(nptr);
            rear=nptr;
        }
        nptr.setLink(front);
    }
    public int delqueue()
    {
        int temp=front.getData();
        if(front==rear)
        {
            front=null;
            rear=null;
        }
        else
        {
            front=front.getLink();
            rear.setLink(front);
        }
        return temp;
    }
    public void show()
    {
        if(isEmpty())
        {
            System.out.println("\nEmpty");
            return;
        }
        System.out.println();
        for(Node ptr=front;ptr.getLink()!=front;ptr=ptr.getLink())
            System.out.print(ptr.getData()+" ");
        System.out.println(rear.getData());
    }
    boolean isEmpty()
    {
        return front==null;
    }
}